protocols/virtualfund/virtualfund.go

```diff
@@ -308,58 +317,69 @@ func (o Objective) Update(event protocols.ObjectiveEvent) (protocols.Objective,
 // Crank inspects the extended state and declares a list of Effects to be executed
 // It's like a state machine transition function where the finite / enumerable state is returned (computed from the extended state)
 // rather than being independent of the extended state; and where there is only one type of event ("the crank") with no data on it at all.
-func (o Objective) Crank(secretKey *[]byte) (protocols.Objective, protocols.SideEffects, protocols.WaitingFor, []protocols.GuaranteeRequest, error) {
+func (o Objective) Crank(secretKey *[]byte) (protocols.Objective, protocols.SideEffects, protocols.WaitingFor, error) {
 	updated := o.clone()
 
 	sideEffects := protocols.SideEffects{}
-	ledgerRequests := []protocols.GuaranteeRequest{}
 	// Input validation
 	if updated.Status != protocols.Approved {
-		return &updated, sideEffects, WaitingForNothing, []protocols.GuaranteeRequest{}, ErrNotApproved
+		return &updated, sideEffects, WaitingForNothing, ErrNotApproved
 	}
 
 	// Prefunding
 
 	if !updated.V.PreFundSignedByMe() {
 		ss, err := updated.V.SignAndAddPrefund(secretKey)
 		if err != nil {
-			return &o, protocols.SideEffects{}, WaitingForNothing, []protocols.GuaranteeRequest{}, err
+			return &o, protocols.SideEffects{}, WaitingForNothing, err
 		}
 		messages := protocols.CreateSignedStateMessages(o.Id(), ss, o.V.MyIndex)
 		sideEffects.MessagesToSend = append(sideEffects.MessagesToSend, messages...)
 	}
 
 	if !updated.V.PreFundComplete() {
-		return &updated, sideEffects, WaitingForCompletePrefund, ledgerRequests, nil
+		return &updated, sideEffects, WaitingForCompletePrefund, nil
 	}
 
 	// Funding
 
-	if !updated.requestedLedgerUpdates {
-		updated.requestedLedgerUpdates = true
-		ledgerRequests = append(ledgerRequests, o.generateGuaranteeRequestSideEffects()...)
+	if !updated.isAlice() && !updated.ToMyLeft.ledgerChannelAffordsExpectedGuarantees() {
```

Comment:
I think there's an opportunity to make the protocol easier to understand with some different names. Eg. Should `isAlice` say `isLeader`?

Reply:
Alice and Bob are (of course) overloaded characters. In the context of the virtualFundObjective Alice and Bob are the left & right participants in V. This is distinct from Leaders and Followers, which are designations meaningful specifically in ledger channels.